### PR TITLE
CMP: add OID decls for CMP-Updates and CMPv3

### PR DESCRIPTION
**Pull request by DDvO**
This adds OID declarations supporting the upcoming CMP-Updates RFC (see https://datatracker.ietf.org/doc/html/draft-ietf-lamps-cmp-updates), which defines extensions to CMPv2 and the new CMP version 3.

This PR is pretty shallow and schematic, in part auto-generated (in `crypto/objects/` and `obj_mac.h`).

### CODE REVIEW

**Comment by DDvO** [comment]
It hit me as a surprise that beta1 likely will be released tomorrow or the day after.
Is there any chance to still bring this in, or certain parts of it?

**Comment by DDvO** [comment]
I'm about to fix the symbol presence and split this up in 2 or 3 PRs...

**Comment by richsalz** [comment]
I do not believe it is necessary that this goes into 3.0

The documents aren't finished yet.  There is still ongoing discussion about some basic points.

**Comment by t-j-h** [comment]
This is too late for 3.0 inclusion and would have needed to be in before the final alpha ...

**Comment by DDvO** [comment]
I see - so removed the tentative milestone.

**Comment by DDvO** [comment]
The **out-of-source-and-install (macos-latest)** CI run exhibts a strange failure of `test_symbol_presence`, which looks like an internal error in a corner case hit by this PR.

File: doc/man3/OSSL_CRMF_MSG_set1_regCtrl_regToken.pod [hunk 1]
```diff
@@ -28,20 +37,32 @@ OSSL_CRMF_CERTID_gen
...
 
 =head1 DESCRIPTION
 
+OSSL_CRMF_MSG_get0_regCtrl_regToken() returns the regToken control
+in the given I<msg>, if present.
+
 OSSL_CRMF_MSG_set1_regCtrl_regToken() sets the regToken control in the given
 I<msg> copying the given I<tok> as value. See RFC 4211, section 6.1.
 
+OSSL_CRMF_MSG_get0_regCtrl_regToken() returns the authenticator control
```
Comment (FdaSilvaYY) [inline doc/man3/OSSL_CRMF_MSG_set1_regCtrl_regToken.pod:63]: nit: `OSSL_CRMF_MSG_get0_regCtrl_regToken` is mentioned 4 times.

Comment (DDvO) [reply doc/man3/OSSL_CRMF_MSG_set1_regCtrl_regToken.pod]: Oops, this was a copy&paste mistake I did in the rush. 
Fixed by making the text generic, which is more succinct and should be better to read.

**Comment by DDvO** [comment]
Rebased in order to get rid of the unrelated CI failure for MacOS.

File: doc/man3/OSSL_CRMF_MSG_set1_regCtrl_regToken.pod (hunk 1 above)
Comment (DDvO) [reply doc/man3/OSSL_CRMF_MSG_set1_regCtrl_regToken.pod]: Thank you @FdaSilvaYY for having an eye on glitches in the documentation in may new PRs.

**Comment by DDvO** [comment]
Re-added the (shrunk) list of CMP-related OIDs from #15780.

**Review by paulidale — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks.

**Pull request closed without merging**